tex/chProgramming.tex

```diff
@@ -1138,9 +1147,10 @@ \subsection{Recursion on natural numbers}\label{ssec:recnat}
 % second argument is 0; but this
 rule thus also covers the case where the
 second argument is non-zero while the first argument is 0: in that
-case, the result of the function is zero.  Another possible view on
-\C{subn} it to see it as a subtraction operation on natural numbers,
-made total by providing a default value in the ``exceptional'' cases.
+case, the result of the function is zero.  We can view
+\C{subn} as a subtraction operation on natural numbers, artificially
+made total by providing a default value in the cases where it would
+normally produce a negative number.
```

Comment:
This is misleading, IMO, because "subtraction operation on natural numbers" cannot "produce a negative number".
I think that both "made total" and "truncated" are good ways to express what it does.
As a CS person, I clearly see total as more natural, but if a reference to truncated subtraction can help the Math inclined reader I suggest we have both.

Reply:
What do you think of the new version?